packages/hooks/src/useQuery.ts

```diff
@@ -6,23 +6,30 @@ import {
 } from '@apollo/react-common';
 import { DocumentNode } from 'graphql';
 
-import { QueryHookOptions, QueryOptions } from './types';
+import { QueryHookOptions, QueryOptions, QueryTuple } from './types';
 import { QueryData } from './data/QueryData';
 import { useDeepMemo } from './utils/useDeepMemo';
 
-export function useQuery<TData = any, TVariables = OperationVariables>(
+export function useQuery<
+  TData = any,
+  TVariables = OperationVariables,
+  TLazy = undefined
+>(
   query: DocumentNode,
-  options?: QueryHookOptions<TData, TVariables>
-): QueryResult<TData, TVariables> {
+  options?: QueryHookOptions<TData, TVariables, TLazy>
+): TLazy extends undefined
+  ? QueryResult<TData, TVariables>
+  : QueryTuple<TData, TVariables> {
```

Comment:
It appears the `QueryTuple` mode would be enabled even if someone passed `useQuery(query, { lazy: false })`, which seems like an argument for `useLazyQuery`?